File: backend/src/main/java/com/example/backend/entity/Nutrition.java
package com.example.backend.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Nutrition {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private double calories; // Total daily calories

    @Column(nullable = false)
    private double proteins; // Grams of protein

    @Column(nullable = false)
    private double carbs; // Grams of carbohydrates

    @Column(nullable = false)
    private double fats; // Grams of fat

    @OneToOne
    @JoinColumn(name = "client_id", referencedColumnName = "id", nullable = false)
    private Client client; // Link to the Client

    public Nutrition(double calories, double protein, double carbs, double fat) {
        this.calories = calories;
        this.proteins = protein;
        this.carbs = carbs;
        this.fats = fat;
    }
}

File: frontend/mobile/Fit/rpstrengh/lib/src/models/nutrition.dart
class Nutrition {
  final int id;
  final double calories;
  final double proteins;
  final double carbs;
  final double fats;
  final int clientId;

  Nutrition({
    required this.id,
    required this.calories,
    required this.proteins,
    required this.carbs,
    required this.fats,
    required this.clientId,
  });

  factory Nutrition.fromJson(Map<String, dynamic> json) {
    return Nutrition(
      id: json['id'] ?? 0,
      calories: json['calories']?.toDouble() ?? 0.0,
      proteins: json['proteins']?.toDouble() ?? 0.0,
      carbs: json['carbs']?.toDouble() ?? 0.0,
      fats: json['fats']?.toDouble() ?? 0.0,
      clientId: json['client']['id'],
    );
  }
}
